lib/embedded_ansible.rb

```diff
@@ -185,6 +192,22 @@ def self.configure_secret_key
   end
   private_class_method :configure_secret_key
 
+  def self.configure_proxy_if_any
+    proxy = VMDB::Util.http_proxy_uri(:embedded_ansible) || VMDB::Util.http_proxy_uri
+    if proxy
+      proxy_settings = <<-EOF
+
+AWX_TASK_ENV['HTTP_PROXY'] = #{proxy}
+AWX_TASK_ENV['HTTPS_PROXY'] = #{proxy}
+
+EOF
+      File.open(SETTING_FILE, 'a') do |file|
+        file.write(proxy_settings)
+      end
```

Comment:
I think this would be cleaner if it were just two calls to `File#write`:

```ruby
File.open(SETTING_FILE, 'a') do |file|
  file.write("AWX_TASK_ENV['HTTP_PROXY'] = #{proxy}\n")
  file.write("AWX_TASK_ENV['HTTPS_PROXY'] = #{proxy}\n")
end
```

Also can we check that these ENV vars are not present in settings.py before writing them? It feels like this would write the config multiple times if the role were removed and re-added.

Reply:
Thanks. Do we need an extra `file.write('\n')` before the first `file.write`. In case `settings.py` doesn't contain a fresh empty line. And what is preferred way to check that ENV vars? By search `AWX_TASK_ENV['HTTP_PROXY']` string in `py` file?

Reply:
> Do we need an extra file.write('\n') before the first file.write. In case settings.py doesn't contain a fresh empty line.

Yeah, that's probably a good idea.

>And what is preferred way to check that ENV vars? By search AWX_TASK_ENV['HTTP_PROXY'] string in py file?

Sounds good to me.